templates/task-sc.yaml

```diff
@@ -0,0 +1,53 @@
+---
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+    name: skopeo-copy
+    labels:
+        app.kubernetes.io/version: 0.0.1
+spec:
+    description: | 
+        Skopeo is a command line tool for working with remote image registries. 
+        The copy command will take care of copying the image from internal.registry to production.registry. If your production registry requires credentials to login in order to push the image, skopeo can handle that as well.
+    params:
+        -   name: SOURCE_REGISTRY
+            type: string
+            description: |
+                Source Registry URL
+        -   name: DESTINATION_REGISTRY
+            type: string
+            description: |
+                Destination Registry URL
+        -   name: SOURCE_AUTH_USERNAME
+            type: string
+            description: |
+                Source Registry Username for authentication
+        -   name: SOURCE_AUTH_PASSWORD
+            type: string
+            description: |
+                Source Registry Password for authentication
+        -   name: DESTINATION_AUTH_USERNAME
+            type: string
+            description: |
+                Destination Registry Username for authentication
+        -   name: DESTINATION_AUTH_PASSWORD
+            type: string
+            description: |
+                Destination Registry Password for authentication
```

Comment:
This type of parameters should be added on the next step, when we start using optional workspaces for advanced authentication cases. Let's focus first on using what Tekton Pipeline provides out of the box.

Reply:
Right, I'm looking into the authentication part!